### ISSUE TITLE
[bug] Disabled `<rh-button>` should still be focusable

### ISSUE THREAD

**Issue by hellogreg** [RedHat-UX/red-hat-design-system#2041]
Currently, `<rh-button>` has a `disabled` attribute which, when set to `true`, adds a `disabled` attribute to its `<button>` element.

This has the effect of making the button unreachable to keyboard users, even when `tabindex` is applied.

In order to make the button focusable, I recommend applying an `aria-disabled="true"` attribute to the `<button>` instead of `disabled`, when `<rh-button>` has the `disabled` attribute applied to it. This also means we can attach actions and/or helpful error messages when anyone (mouse or keyboard, visual or non-visual) tries to interact with it: "You must update at least one option before re-saving" or some such.

Basically...
- Using `aria-disabled` allows screen reader users to focus on it.
- The button can still _look_ disabled to visual users. 
- We can add styles like `cursor: not-allowed`, if desired.
- JS can be used to prevent clicks, if desired.

By doing the above, we wind up with what the `<button disabled>` attribute should be by default, imo.

Might want to pair this issue with #1884, which also relates to disabled `rh-button`.

If you feel like reading, here's a [good article on this topic](https://css-tricks.com/making-disabled-buttons-more-inclusive/#aa-aria-to-the-rescue).
